cachito/web/models.py

```diff
@@ -171,6 +180,7 @@ def to_json(self):
             'state_history': states,
             'user': user,
             'environment_variables': env_vars_json,
+            'flags': [flag.to_json() for flag in self.flags if flag.active],
```

Comment:
This doesn't look right to me. Don't we want to show the flags that were used historically even if they were disabled?

Reply:
I'm struggling to find the right balance here. If we display them, we have a transparent way of showing historical results. However, users can't "replay" the same request. Knowledge of what's active is required.

For auditing purposes, we could still inspect the database for all the flags used on any request, but this requires DB access.

We could represent the active state of a flag in the response but that seems a bit verbose, and potentially unnecessary, to me.

Reply:
@lcarva, I think we should show flags as they were historically. If the user replays a request and the flags that are disabled are omitted, the old request isn't actually reproduced. This is misleading to the user in my opinion. I think we should show an error message in this case stating that the following flags in replayed request are not valid.

Reply:
Consider the case where we want to roll out a feature gradually. This will be done by having some users setting a flag. Eventually, that feature is main stream and all the requests will behave as if the flag is set. At this point, we should mark the flag as inactive.

It seems odd to me to make those previous requests not "replayable". However, I do see the point that hiding those flags may make things more complicated.

What if we always show the used flags regardless if they're no longer active, as you suggested, but change how we handle processing them for new requests.

Going back to the example I mentioned earlier, it would be a pretty bad user experience if we decide to deem a flag no longer necessary and mark it as inactive, only to break requests for those early adopters using the flag. In the POST processing, we could filter out inactive flags from the request, and continue to reject invalid flags. This also addresses the replaying scenario. To be clear, I don't think the inactive flags should be added to the request, but could be silently ignored.

Reply:
@lcarva, I had misunderstood the point of the "active" column on the flag then. I had assumed that a flag would be disabled if we didn't want to allow the use of the flag anymore. For instance, we tried a new feature and decided it wasn't a good idea and wanted to disable it.

Your approach works for me. @yashvardhannanavati, could you make the change so that the flags are not filtered? Since we don't allow a request to be "replayable" yet, there's nothing to do for that part of the feature that was described.

Reply:
+1 I'll make that change and submit a separate PR.